### ISSUE TITLE
Whitelist regex includes listenpath even when  "strip_listen_path": true

### ISSUE THREAD

**Issue by jakub-bochenski** [TykTechnologies/tyk#5595]
**Branch/Environment/Version**
- Branch/Version: v5.1
- Environment: On-prem

**Describe the bug**
I've whitelisted a single request to be `[0-9]+`. But requests with non numeric component are still accepted

EDIT: It appears that to get what I want I need to specify      "path": "^/search/v1/[0-9]+$" as regex.

This seems bad, as it forces me to repeat the listen_path and version prefix in every regex.

This is both bloat and makes it hard to reuse the statements for a different listen path or version

**Reproduction steps**
Define an API
```
{
  "name": "Search",
  "api_id": "search",
  "org_id": "default",
  "definition": {
    "location": "url"
  },
  "use_keyless": true,
  "version_data": {
    "not_versioned": false,
    "versions": {
      "v1": {
        "name": "v1",
        "use_extended_paths": true,
        "extended_paths": {
          "url_rewrites": [
            {
              "method": "GET",
              "path": "{project-id}",
              "match_pattern": "(\\d+)",
              "rewrite_to": "gateway?projectId=$1"
            }
          ],
          "white_list": [
            {
              "method": "GET",
              "path": "[0-9]+"
            }
          ]
        }
      }
    }
  },
  "proxy": {
    "listen_path": "/search/",
    "target_url": "https://httpbin.org/anything/",
    "strip_listen_path": true
  }
}
```
Make a request `curl http://localhost:8080/search/v1/abcd`

**Actual behavior**
The request is proxied to upstream.

**Expected behavior**
Request is rejected

**Logs (debug mode or log file):**
```
tyk-gateway_1  | time="Oct 05 16:57:23" level=debug msg=Started api_id=search api_name=Search mw=VersionCheck org_id=default origin=172.19.0.1 path=/search/v1/abcd ts=1696525043860407291
tyk-gateway_1  | time="Oct 05 16:57:23" level=debug msg=Finished api_id=search api_name=Search code=200 mw=VersionCheck ns=49905 org_id=default origin=172.19.0.1 path=/search/v1/abcd
tyk-gateway_1  | time="Oct 05 16:57:23" level=debug msg=Started api_id=search api_name=Search mw=RateCheckMW org_id=default origin=172.19.0.1 path=/search/v1/abcd ts=1696525043860474320
tyk-gateway_1  | time="Oct 05 16:57:23" level=debug msg=Finished api_id=search api_name=Search code=200 mw=RateCheckMW ns=22318 org_id=default origin=172.19.0.1 path=/search/v1/abcd
tyk-gateway_1  | time="Oct 05 16:57:23" level=debug msg=Started api_id=search api_name=Search mw=URLRewriteMiddleware org_id=default origin=172.19.0.1 path=/search/v1/abcd ts=1696525043860525183
tyk-gateway_1  | time="Oct 05 16:57:23" level=debug msg="Rewriter active"
tyk-gateway_1  | time="Oct 05 16:57:23" level=debug msg=/search/v1/abcd
tyk-gateway_1  | time="Oct 05 16:57:23" level=debug msg="Inbound path: /search/v1/abcd"
tyk-gateway_1  | time="Oct 05 16:57:23" level=debug msg="Rewriter checking matches, len is: 1"
tyk-gateway_1  | time="Oct 05 16:57:23" level=debug msg="[[1 1]]"
tyk-gateway_1  | time="Oct 05 16:57:23" level=debug msg="[[$1]]"
tyk-gateway_1  | time="Oct 05 16:57:23" level=debug msg="URL Re-written from: /search/v1/abcd"
tyk-gateway_1  | time="Oct 05 16:57:23" level=debug msg="URL Re-written to: gateway?projectId=1"
tyk-gateway_1  | time="Oct 05 16:57:23" level=debug msg=Finished api_id=search api_name=Search code=200 mw=URLRewriteMiddleware ns=85262 org_id=default origin=172.19.0.1 path=/search/v1/abcd
tyk-gateway_1  | time="Oct 05 16:57:23" level=debug msg="Started proxy"
tyk-gateway_1  | time="Oct 05 16:57:23" level=debug msg="Stripping proxy listen path: /search/"
tyk-gateway_1  | time="Oct 05 16:57:23" level=debug msg="Upstream path is: /gateway"
tyk-gateway_1  | time="Oct 05 16:57:23" level=debug msg=Started api_id=search api_name=Search mw=ReverseProxy org_id=default ts=1696525043860632828
tyk-gateway_1  | time="Oct 05 16:57:23" level=debug msg="Upstream request URL: /gateway?projectId=1" api_id=search api_name=Search mw=ReverseProxy org_id=default
tyk-gateway_1  | time="Oct 05 16:57:23" level=debug msg="Outbound request URL: https://httpbin.org/anything/gateway?projectId=1" api_id=search api_name=Search mw=ReverseProxy org_id=default
tyk-gateway_1  | time="Oct 05 16:57:23" level=debug msg="Creating new transport" api_id=search api_name=Search mw=ReverseProxy org_id=default
tyk-gateway_1  | time="Oct 05 16:57:23" level=debug msg="Out request url: https://httpbin.org/anything/gateway?projectId=1" api_id=search api_name=Search mw=ReverseProxy org_id=default
tyk-gateway_1  | time="Oct 05 16:57:24" level=debug msg=Finished api_id=search api_name=Search mw=ReverseProxy ns=674801549 org_id=default

```

**Comment by jakub-bochenski** [comment]
It seems that `\d+` in path is working like `.+`
```
tyk-gateway_1  | time="Oct 05 15:45:02" level=debug msg=Started api_id=search api_name=Search mw=VersionCheck org_id=default origin=172.19.0.1 path=/search/v1/abc ts=1696520702660512538
tyk-gateway_1  | time="Oct 05 15:45:02" level=debug msg=Finished api_id=search api_name=Search code=200 mw=VersionCheck ns=57338 org_id=default origin=172.19.0.1 path=/search/v1/abc
```
What gives?

Only by changing the path to sth like `"(\\d+)/foo" I can observe requests being rejected

**Comment by jakub-bochenski** [comment]
I think there's something wrong with version handing here, notice the version component is not removed
```
tyk-gateway_1  | time="Oct 05 16:47:48" level=debug msg="Stripping proxy listen path: /search/"
tyk-gateway_1  | time="Oct 05 16:47:48" level=debug msg="Upstream path is: /v1/abcd"
```

**Comment by jakub-bochenski** [comment]
Also notice the rewrite is somehow capturing `abcd` into a `(\d+)` capturing group as `1`

```
tyk-gateway_1  | time="Oct 05 16:57:23" level=debug msg="Rewriter checking matches, len is: 1"
tyk-gateway_1  | time="Oct 05 16:57:23" level=debug msg="[[1 1]]"
tyk-gateway_1  | time="Oct 05 16:57:23" level=debug msg="[[$1]]"
tyk-gateway_1  | time="Oct 05 16:57:23" level=debug msg="URL Re-written from: /search/v1/abcd"
tyk-gateway_1  | time="Oct 05 16:57:23" level=debug msg="URL Re-written to: gateway?projectId=1"
```

**Comment by jakub-bochenski** [comment]
It appears that to get what I want I need to specify `     "path": "^/search/v1/[0-9]+$"` as regex.

This seems bad, as it forces me to repeat the listen_path and version prefix in every regex.

This is both bloat and makes it hard to reuse the statements for a different listen path or version

**Comment by jakub-bochenski** [comment]
Closed in favour of #5601
